### ISSUE TITLE
testing SS with data from an operating model?

### ISSUE THREAD

**Issue by k-doering-NOAA** [nmfs-ost/ss3-source-code#174]
Thanks to Bai Li and Matthew Supernaw who are setting up tests of MAS using data from an operating model.

We could experiment with testing SS with data set(s) from an operating model. In this case, we would know what the true values are supposed to be, so could check that SS is getting reasonable results for all the values.

The positive aspect is that if the operating model is not based on SS, we can be a little more sure that SS is producing the "correct" results (as long as we trust that the original operating model is correctly set up). This is in comparison to our current checks based on previous runs of SS, that only checks that results haven't changed since the last "acceptable" model run.

The difficulty is it may be more challenging to figure out what tolerances to accept and if we want to run the model multiple times, then run times could be long.

This might be something like basically setting up Bai's model comparison project work to run more frequently with the current ss version. Maybe this is something we would want to run less than every commit if it takes a long time to run (maybe on a schedule, for example weekly or monthly? Or maybe only before a release?)

I think exploring a test like this could provide added value.

**Comment by Rick-Methot-NOAA** [comment]
I see some merit in having such a data set in our testing repertoire, but it would be a very simple test (no lengths or any of the SS bells & whistles).  Also, Bai's model comparison project ended up finding a difference between SS formulation and BAM's formulation.  Neither was wrong, but some transformation was needed to correctly compare.  I think I see more merit in using ss3sim to generate some known data sets.

**Comment by k-doering-NOAA** [comment]
Yes, the ss3sim way is another approach that could be helpful; the issue there is because it is dependent on SS, if the bug is in the operating model it will also be in the estimation model. But we would have the flexibility to easily generate complex setups.

**Comment by k-doering-NOAA** [comment]
@Bai-Li-NOAA do you see value in this? I just came across this issue. I think it may only be worth setting up if there are additional features added into the OM after the model comparison project. I thought you and Matthew were working on adding some spatial capabilities, but I could not be remembering correctly...

**Comment by Rick-Methot-NOAA** [comment]
This has largely been accomplished already.  Closing the issue.
